src/components/common/ColonyHome/ColonyUpgrade/ColonyUpgrade.tsx

```diff
@@ -34,18 +39,16 @@ const MSG = defineMessages({
 
 const ColonyUpgrade = () => {
   const { colony } = useColonyContext();
-
-  const openUpgradeVersionDialog = useDialog(NetworkContractUpgradeDialog);
-  const { version: networkVersion } = useNetworkContracts();
+  const { colonyContractVersion } = useColonyContractVersion();
+  const openUpgradeColonyDialog = useDialog(NetworkContractUpgradeDialog);
   const { user, wallet } = useAppContext();
-
-  const handleUpgradeColony = useCallback(
-    () =>
-      openUpgradeVersionDialog({
-        colony,
-      }),
-    [colony, openUpgradeVersionDialog],
-  );
+  const enabledExtensionData = useEnabledExtensions();
+  const handleUpgradeColony = () =>
+    colony &&
```

Comment:
You don't need to check for `colony` object existence, the `useColonyContext` already does that. You won't get to this point if `colony` is not set

Reply:
I agree but `useColonyContext` is returning `colony` as `ColonyFragment | undefined` anyways, so if I don't add these checks then TS just marks this as an error.

As we've mentioned in other PRs, I think we will need to refactor `useColonyContext` to actually return the correct type, or in the worst-case scenario, recast it inside the hook before returning it.